packages/contracts/src/contracts/current/protocol/Exchange/LibOrder.sol

```diff
@@ -65,15 +65,15 @@ contract LibOrder {
                 address(this),
                 order.makerAddress,
                 order.takerAddress,
-                order.makerTokenAddress,
-                order.takerTokenAddress,
                 order.feeRecipientAddress,
                 order.makerTokenAmount,
                 order.takerTokenAmount,
                 order.makerFee,
                 order.takerFee,
                 order.expirationTimeSeconds,
-                order.salt
+                order.salt,
+                order.makerAssetData,
+                order.takerAssetData
```

Comment:
I find it strange that the maker sets the `asset transfer proxy` for the taker. Locking in the `takerTokenAddress` and the `assetId` (in the case of 721 makes sense). But it definitely feels weird that a maker decides to use the OLD proxy, forcing a Taker (who has set an allowance on the NEW proxy) to do the same.

I think it's good that we've tried to re-use the old proxy to avoid users having to re-set allowances. I think it works in theory, but will it actually work in practice, or will it become a more confusing user onboarding?

Reply:
I don't think we should be doin g any decoding in the exchange contract, since the point is to have `makerAssetData` and `takerAssetData` be unstructured (except for the first byte).

I also think that it generally makes sense to have the maker decide the assetProxy for the taker. The example I like to think of here is if we implemented a deposit contract as a proxy, the maker probably would only want the taker to use the deposit contract if they too were using it. However, if the proxies are not sufficiently differentiated then it doesn't really make sense (like the old/new ERC20 proxies). Maybe there's a way for the maker to specify multiple acceptable proxies? It increases complexity for sure, but the flexibility seems like it could be worth it.

Reply:
We chatted about this offline, and decided the old ERC20 proxy will be removed for simplicity.